ckanext/canada/logic.py

```diff
@@ -123,19 +124,27 @@ def activity_list_from_user_since(context, data_dict):
         since_time, limit,user_id)
     return model_dictize.activity_list_dictize(activity_objects, context)
 
+
 def _changed_packages_activity_list_since(since, limit):
-    '''Return the site-wide stream of changed package activities since a given
-    date.
+    '''Return package_ids and last activity date of changed package
+    activities since a given date.
 
-    This activity stream includes recent 'new package', 'changed package' and
-    'deleted package' activities for the whole site.
+    This activity stream includes recent 'new package', 'changed package',
+    'deleted package', 'new resource view', 'changed resource view' and
+    'deleted resource view' activities for the whole site.
 
     '''
-    q = model.activity._changed_packages_activity_query()
-    q = q.order_by(model.Activity.timestamp)
+    q = model.Session.query(model.Activity.object_id.label('object_id'),
+                            func.max(model.Activity.timestamp).label(
```

Comment:
you're taking the max timestamp but you need the min in order to paginate, right?